app/models/course.rb

```diff
@@ -35,7 +35,8 @@ class Course < ActiveRecord::Base
                                 source: :actable, source_type: Course::LessonPlan::Event.name
   has_many :material_folders, class_name: Course::Material::Folder.name, inverse_of: :course,
                               dependent: :destroy
-
+  has_one :root_folder, -> { where(parent_id: nil) },
```

Comment:
...And we might need a conditional unique index for current implementation as well, since consistency_fail failed.